lib/AST/Decl.cpp

```diff
@@ -551,6 +551,12 @@ bool Decl::isWeakImported(ModuleDecl *fromModule) const {
   if (getAttrs().hasAttribute<WeakLinkedAttr>())
     return true;
 
+  if (auto *accessor = dyn_cast<AccessorDecl>(this))
+    return accessor->getStorage()->isWeakImported(fromModule);
+
+  if (auto *dtor = dyn_cast<DestructorDecl>(this))
+    return cast<ClassDecl>(dtor->getDeclContext())->isWeakImported(fromModule);
```

Comment:
I don't think we need this one. It should be covered by availability when we implement that, but otherwise you'd have to apply this rule to every member of every decl.